File: android-app/app/src/main/java/com/imran/receptionist/call/PhoneNormalizer.kt
package com.imran.receptionist.call

object PhoneNormalizer {
    fun normalize(phoneNumber: String): String {
        // Remove all non-digit characters
        val digits = phoneNumber.replace(Regex("\\D"), "")

        // If 10 digits, assume India (add country code)
        if (digits.length == 10) {
            return "91$digits"
        }

        // Return as-is if already in international format or unclear
        return digits
    }
}

File: backend/src/utils/phoneNormalizer.js
export function normalizePhoneNumber(phoneNumber) {
  // Remove all non-digit characters
  let normalized = phoneNumber.replace(/\D/g, '');

  // If it starts with country code, keep it
  // Otherwise, assume it's just the number without country code
  if (normalized.length > 10 && normalized.length <= 15) {
    // Looks like international format
    return normalized;
  }

  if (normalized.length === 10) {
    // Assume India if 10 digits
    // You can modify this based on your region
    return '91' + normalized;
  }

  // Return as-is if uncertain
  return normalized;
}
